core/modules/file/file.module

```diff
@@ -2929,42 +2896,72 @@ function file_stream_wrappers_alter(&$wrappers) {
  * Implements hook_file_access().
  */
 function file_file_access($op, $file, $account) {
-  // If the file URI is invalid, deny access.
-  if (is_object($file) && isset($file->uri) && !file_valid_uri($file->uri)) {
-    if(isset($file->is_new) && $file->is_new == true && user_access('create files', $account)) {
-      return FILE_ACCESS_ALLOW;
-    }
-    return FILE_ACCESS_DENY;
-  }
 
   if ($op == 'create') {
+    // @todo File::createAccess?
```

Comment:
Regarding this commented line above: We should create a static method File::createAccess() - and I'm aware that there is an issue for that. 

However, to mirror the node access workflow, the invocation of `File::createAccess()` should exist only within `file_access()` - not inside this hook `file_file_access()`.  If we mirror Node, `File::createAccess()` will invoke this hook, so we don't want to create an infinite loop.

Reply:
And the same question again... :laughing: (you're not the first one who asked that).

The PR for File::createAccess() is [over there](https://github.com/backdrop/backdrop/pull/3929). I'm tempted to incorporate that here and close the other PR and issue.

Any concerns regarding that?

Reply:
It'd be a good idea to close that and incorporate here.

Note I said I was aware of the issue. My concern (the reason for my comment) was seeing this commented line ("// @todo File::createAccess?")  inside file_file_access() - which kind of suggested to me that the method `File::createAcess()` would be called here,  instead of in file_access() the invocation to `File::createAcess()` belongs.   

Sorry if that was not your intention.

Reply:
I could write that I understand what you mean - but then I'd be lying. :grinning:

Reply:
That's OK. Better for me to comment on actual code rather than on commented lines. Onward.